libqtile/backend/wayland/window.py

```diff
@@ -1415,6 +1420,13 @@ def _on_set_class(self, _listener: Listener, _data: Any) -> None:
         self._wm_class = self.surface.wm_class
         self.ftm_handle.set_app_id(self._wm_class or "")
 
+    def _on_set_geometry(self, _listener: Listener, _data: Any) -> None:
+        logger.debug("Signal: xstatic set_geometry")
+        self.x = self.surface.x
```

Comment:
It looks like we aren't respecting a new width or height. Should we also do that?

It would also be good to use `self.place` to reset the position rather than having some lines duplicated here - `self._find_outputs` is needed too.

Reply:
Ah sorry I wasn't clear - lines 1426-1429 aren't needed because the values passed into `place` will be used. You also need to pass in `0` as the borderwidth.

Reply:
As stated earlier, the bug still persists if I don't explicitly mention self.x = self.surface.x for some odd reason that I don't currently understand.

Reply:
Ah I see. It's because your place() call fails because it needs the borderwidth argument, which is required. I think if you pass that then you won't need the manual setters for `self.x` and `self.y`.

Reply:
> Ah I see. It's because your place() call fails because it needs the borderwidth argument, which is required. I think if you pass that then you won't need the manual setters for `self.x` and `self.y`.

already thought of that and created the following line:
```py
self.place(self.surface.x, self.surface.y, geometry.width, geometry.height, 0, None)
```
But oh well it still doesn't work. I'll look into this when i get some time.

Reply:
> logger.exception

Hm I didn't try a `try/except` block before. But sad news, still doesn't work ahaha just silently fails.

I'll push my progress so maybe you can take a look?